bundles/org.openhab.core.io.rest.sitemap/src/main/java/org/openhab/core/io/rest/sitemap/SitemapSubscriptionService.java

```diff
@@ -215,18 +217,39 @@ public boolean exists(String subscriptionId) {
      * @return the name of the current sitemap or null if no sitemap is currently set for the subscription
      */
     public @Nullable String getSitemapName(String subscriptionId) {
-        String sitemapWithPageId = pageOfSubscription.get(subscriptionId);
+        String sitemapWithPageId = scopeOfSubscription.get(subscriptionId);
         return (sitemapWithPageId == null) ? null : extractSitemapName(sitemapWithPageId);
     }
 
     private String extractSitemapName(String sitemapWithPageId) {
         return sitemapWithPageId.split(SITEMAP_PAGE_SEPARATOR)[0];
     }
 
+    private boolean isPageListener(String sitemapWithPageId) {
+        return sitemapWithPageId.contains(SITEMAP_PAGE_SEPARATOR);
+    }
+
     private String extractPageId(String sitemapWithPageId) {
         return sitemapWithPageId.split(SITEMAP_PAGE_SEPARATOR)[1];
     }
 
+    public void unsetPageId(String subscriptionId, String sitemapName) {
```

Comment:
I don't think you need this method if you allow `pageId` to be `null` and act accordingly. This also simplifies code in the `SitemapResource`